packages/server/api/src/app/ee/custom-domains/custom-domain.service.ts

```diff
@@ -1,23 +1,60 @@
 import { databaseConnection } from '../../database/database-connection'
 import { buildPaginator } from '../../helper/pagination/build-paginator'
 import { paginationHelper } from '../../helper/pagination/pagination-utils'
+import { getEdition } from '../../helper/secret-helper'
+import { cloudflareHostnameServices } from './cloudflare-api.service'
 import { CustomDomainEntity } from './custom-domain.entity'
 import {
     CustomDomain,
     CustomDomainStatus,
     ListCustomDomainsRequest,
 } from '@activepieces/ee-shared'
-import { apId, SeekPage } from '@activepieces/shared'
+import { logger } from '@activepieces/server-shared'
+import { ActivepiecesError, ApEdition, apId, ErrorCode, SeekPage } from '@activepieces/shared'
+
+export type SSLParams = {
+    bundleMethod: 'ubiquitous' | 'optimal' | 'force'
+    certificateAuthority: 'lets_encrypt'
+    customCertificate: string
+    customKey: string
+    method: 'txt'
+}
 
 const customDomainRepo =
-  databaseConnection.getRepository<CustomDomain>(CustomDomainEntity)
+    databaseConnection.getRepository<CustomDomain>(CustomDomainEntity)
 
 export const customDomainService = {
     async delete(request: { id: string, platformId: string }): Promise<void> {
-        await customDomainRepo.delete({
-            id: request.id,
-            platformId: request.platformId,
-        })
+        try {
+            const edition = getEdition()
+            if (edition === ApEdition.CLOUD) {
```

Comment:
We need to allow people other than cloud to delete custom domain but we just want to avoid calling cloudflare

Reply:
Yup, it checks if edition is cloud, if yes then it will start deleting from cloudflare then from DB, otherwise it will skip cloudflare and go delete from DB directly